include/zq/context.hpp

```diff
@@ -34,19 +34,25 @@ namespace zq {
   };
 
   class Context {
-    // The actual constructor
+    // The actual constructors
     friend tl::expected<Context, ErrMsg> mk_context(
         ContextOptions auto const& options) noexcept;
 
+    friend tl::expected<Context, std::runtime_error> mk_context(
+        void* ctx) noexcept;
+
     void* z_ctx{nullptr};
 
-    Context(void* ctx) noexcept : z_ctx{ctx} {};
+    Context(void* ctx, bool owns_ctx = true) noexcept : z_ctx{ctx}, m_owns_ctx{owns_ctx} {};
```

Comment:
Do we need the `owns_ctx` as a constructor parameter? I think the constructor that takes a native context could set `m_owns_ctx` to `false` always.

Reply:
There is only one constructor which takes a pointer to a native context, which is used by the original `mk_context` functions.